pkg/providers/vsphere/vsphere.go

```diff
@@ -1226,37 +1238,38 @@ func (p *vsphereProvider) RunPostControlPlaneUpgrade(ctx context.Context, oldClu
+		// it's never refreshed, it's only created once
+		// In new versions of the capv provider, this is not managed by the controller directly anymore but just with a ClusterResourceSet
+		// Which means that adding update capabilities to the ClusterResourceSet controller and updating our capv provider version will solve this problem
+		err = p.Retrier.Retry(
+			func() error {
+				return p.providerKubectlClient.SetDaemonSetImage(
+					ctx,
+					workloadCluster.KubeconfigFile,
+					cloudControllerDaemonSetName,
+					cloudControllerDaemonSetNamespace,
+					cloudControllerDaemonSetContainerName,
+					clusterSpec.VersionsBundle.VSphere.Manager.VersionedImage(),
+				)
+			},
+		)
+		if err != nil {
+			return fmt.Errorf("failed updating the VSphere cloud controller manager daemonset post upgrade: %v", err)
+		}
 
-	// Step 2: Patch DaemonSet vsphere-cloud-controller-manager in namespace kube-system
-	// More unfortunate stuff. This DaemonSet is created by the capv controller. However, even if it's part of the reconciliation step
-	// it's never refreshed, it's only created once
-	// In new versions of the capv provider, this is not managed by the controller directly anymore but just with a ClusterResourceSet
-	// Which means that adding update capabilities to the ClusterResourceSet controller and updating our capv provider version will solve this problem
-	err = p.Retrier.Retry(
-		func() error {
-			return p.providerKubectlClient.SetDaemonSetImage(
-				ctx,
-				workloadCluster.KubeconfigFile,
-				cloudControllerDaemonSetName,
-				cloudControllerDaemonSetNamespace,
-				cloudControllerDaemonSetContainerName,
-				clusterSpec.VersionsBundle.VSphere.Manager.VersionedImage(),
-			)
-		},
-	)
-	if err != nil {
-		return fmt.Errorf("failed updating the VSphere cloud controller manager daemonset post upgrade: %v", err)
-	}
-
-	// Step 2: Patch DaemonSet vsphere-cloud-controller-manager in namespace kube-system with toleration values for the taints provided in
-	// the cluster spec file
-	err = p.Retrier.Retry(
-		func() error {
-			return p.providerKubectlClient.ApplyTolerationsFromTaintsToDaemonSet(ctx, oldClusterSpec.Cluster.Spec.ControlPlaneConfiguration.Taints, clusterSpec.Cluster.Spec.ControlPlaneConfiguration.Taints, cloudControllerDaemonSetContainerName, workloadCluster.KubeconfigFile)
-		},
-	)
-	if err != nil {
-		return fmt.Errorf("failed to apply tolerations on VSphere cloud controller manager daemonset post upgrade: %v", err)
+		// Step 2: Patch DaemonSet vsphere-cloud-controller-manager in namespace kube-system with toleration values for the taints provided in
+		// the cluster spec file
+		err = p.Retrier.Retry(
+			func() error {
+				return p.providerKubectlClient.ApplyTolerationsFromTaintsToDaemonSet(ctx, oldClusterSpec.Cluster.Spec.ControlPlaneConfiguration.Taints, clusterSpec.Cluster.Spec.ControlPlaneConfiguration.Taints, cloudControllerDaemonSetContainerName, workloadCluster.KubeconfigFile)
```

Comment:
Do we need this ? I can see that the older taints are being hard coded.

Reply:
This is just preserving the older logic